### PR TITLE
Migrate to hatch

### PR DESCRIPTION
**Pull request by rsnyman**
- Migrate to hatch
- Update Python requirements
- Update pre-commit config

### CODE REVIEW

File: .flake8 [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[flake8]
+ignore = E128,E811,W503,E203
+max-line-length = 100
```
Comment (quarckster) [inline .flake8:3]: Please move this config into `pyproject.toml`

**Comment by quarckster** [comment]
Fixes #36

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+dev = [
+    "pytest-subtests",
+    "pytest-xdist",
+]
+
+[tool.hatch.version]
+source = "vcs"
+
+[tool.hatch.build.hooks.vcs]
+version-file = "pytest_ibutsu/_version.py"
```
[the remark this answers is not in the document]
Comment (rsnyman) [reply pyproject.toml:46]: Yeah, that file is generated by setuptools_scm, and should not be part of VCS. But I agree, we don't actually use it yet, so we can remove the file.

File: .flake8 (hunk 1 above)
Comment (rsnyman) [reply .flake8:3]: Can't. Flake8 doesn't support pyproject.toml.

Comment (quarckster) [reply .flake8:3]: What the shame! Do you think it's worth moving it to `.pre-commit-config.yaml` then?

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    "pytest-xdist",
+]
+
+[tool.hatch.version]
+source = "vcs"
+
+[tool.hatch.build.targets.sdist]
+include = [
+    "/src",
+]
```
Comment (quarckster) [inline pyproject.toml:48]: Please add pytest entry point

```toml
[project.entry-points.pytest11]
ibutsu = "pytest_ibutsu.pytest_plugin"
```

**Comment by quarckster** [comment]
@rsnyman `MANIFEST.IN` can be deleted as well

File: .flake8 (hunk 1 above)
Comment (rsnyman) [reply .flake8]: Hmm, the only side-effect of that is that my local flake8 hooks in vim now will complain about this stuff. Ah well.

**Pull request merged**